controllers/profiles/followAnonUser.js

```diff
@@ -12,9 +12,26 @@ const UsersFunction = require('../../databases/functions/users');
 const {sendFollowMainFeedF2} = require('../../services/queue/mainFeedF2');
 
 module.exports = async (req, res) => {
-  const {user_id_followed, follow_source} = req.body;
+  let {user_id_followed, follow_source, post_id, comment_id} = req.body;
   const {user} = req;
 
+  if (follow_source === 'post') {
+    const post = await Getstream.feed.getPlainFeedById(post_id);
+    if (!post) return ErrorResponse.e403(res, 'Post id not found');
+
+    user_id_followed = post?.actor?.id;
+  } else if (follow_source === 'comment') {
+    try {
+      const reaction = await Getstream.feed.getReactionById(comment_id);
+
+      if (!reaction) return ErrorResponse.e403(res, 'Comment id not found');
+
+      user_id_followed = reaction?.user?.id;
+    } catch (error) {
+      return ErrorResponse.e403(res, 'Comment id not found');
+    }
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a potential security issue. The `user_id_followed` is being set based on the `post_id` or `comment_id` provided in the request body without any validation of ownership or permissions. This could allow a malicious user to follow any other user by simply knowing their post or comment ID.

To mitigate this, you should add checks to ensure that the current user has the necessary permissions to follow the user associated with the given post or comment ID. This could be done by adding a method in your service layer that checks if the current user can follow the user associated with the post or comment ID.

```diff
  if (follow_source === 'post') {
    const post = await Getstream.feed.getPlainFeedById(post_id);
    if (!post) return ErrorResponse.e403(res, 'Post id not found');

+    if (!canFollow(user, post.actor.id)) return ErrorResponse.e403(res, 'You do not have permission to follow this user');
    user_id_followed = post?.actor?.id;
  } else if (follow_source === 'comment') {
    try {
      const reaction = await Getstream.feed.getReactionById(comment_id);

      if (!reaction) return ErrorResponse.e403(res, 'Comment id not found');
+      if (!canFollow(user, reaction.user.id)) return ErrorResponse.e403(res, 'You do not have permission to follow this user');
      user_id_followed = reaction?.user?.id;
    } catch (error) {
      return ErrorResponse.e403(res, 'Comment id not found');
    }
  }
```

Please replace `canFollow` with your actual method for checking permissions.


<!-- This is an auto-generated comment by OSS CodeRabbit -->